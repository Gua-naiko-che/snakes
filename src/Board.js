import React from 'react';
import './Board.css';

function Board({ size }) {
  const indexes = [...Array(size).keys()];

  return (
    <table className="board">{renderRows()}</table>
  );

  function renderRows() {
    return indexes.map(renderRow);
  }

  function renderRow(rowIndex) {
    const cells = indexes.map(colIndex => renderCell(rowIndex, colIndex));
    return <tr key={rowIndex}>{cells}</tr>;
  }

  function renderCell(rowIndex, colIndex) {
    return <td className={getClass(rowIndex, colIndex)} key={colIndex}></td>;
  }

  function getClass(rowIndex, colIndex) {
    return rowIndex * colIndex % 2 === 0 ? "snake" : "";
  }
}

export default Board;
