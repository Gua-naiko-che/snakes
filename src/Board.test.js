import React from 'react';
import Board from './Board';
import renderer from 'react-test-renderer';

it('renders correctly', () => {

  const tree = renderer
    .create(<Board />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});