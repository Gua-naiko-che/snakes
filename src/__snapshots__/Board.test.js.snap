// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="board"
>
  <tr>
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
  </tr>
  <tr>
    <td
      className="snake"
    />
    <td
      className=""
    />
    <td
      className="snake"
    />
    <td
      className=""
    />
    <td
      className="snake"
    />
  </tr>
  <tr>
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
  </tr>
  <tr>
    <td
      className="snake"
    />
    <td
      className=""
    />
    <td
      className="snake"
    />
    <td
      className=""
    />
    <td
      className="snake"
    />
  </tr>
  <tr>
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
    <td
      className="snake"
    />
  </tr>
</table>
`;
